### PR TITLE
Clear after commit procs after they are called 

### PR DESCRIPTION
**Pull request by jcredding**
This updates the `AfterCommit` mixins to clear their after commit
procs after they call them. This avoids strange scenarios when the
same instance of a model is saved multiple times. Previously, any
after commit procs that were called would still be configured on
the instance of the model. So if the model was saved again, it
would called again. Though this could be the desired effect, its
more likely the callbacks were added as part of the original save
so they shouldn't called again with the next save.

This adds a `clear_after_commit_procs` to the `RecordProcsMethods`
mixin so that both the `Record` and `FakeRecord` have the method.
The method takes an optional list of callback types which allows
clearing specific callback types or if no callback types are passed
it will clear all of them.

This also adds a `called_after_commit_procs` method to the
`RecordProcsMethods` mixin. This allows reading the procs that have
been called. This is specifically useful for testing that an after
commit proc was added and called.

Finally, this also updates the `Record` and `FakeRecord` mixins to
use the clear after commit procs method after they call their procs
and store their called after commit procs.

@kellyredding - Ready for review.

### CODE REVIEW

**Comment by kellyredding** [comment]
@jcredding we discussed making this automatic in the records and fake records.  I'll look again once that change is in place.

**Comment by jcredding** [comment]
@kellyredding - I've reworked this to automatically clear after commit procs after they are called and to track the after commit procs that it does call for testing.

**Comment by kellyredding** [comment]
@jcredding "... *would called* again." in commit?

File: lib/mr/after_commit/record.rb [hunk 1]
```diff
@@ -26,17 +26,19 @@ module InstanceMethods
       # ActiveRecord runs `after_save` after `after_create`, so run the save
       # procs after the create procs
       def mr_after_commit_run_procs_for_create
```
Comment (kellyredding) [inline lib/mr/after_commit/record.rb]: @jcredding maybe rename these methods to "call" instead of "run"?

**Comment by kellyredding** [comment]
@jcredding this looks great - I like this much better now.  One more minor comment - fix that and I'm good here 👎

**Pull request merged**